src/hera/workflows/service.py

```diff
@@ -97,8 +99,17 @@ def list_archived_workflows(
 
         if resp.ok:
             return WorkflowList(**resp.json())
-        else:
-            raise Exception(f"Server returned status code {resp.status_code} with error: {resp.json()}")
+
+        try:
```

Comment:
@flaviuvadan we seem to be repeating these lines everywhere. Should we move these under the function call as well?

Reply:
I think what you and Elliot suggested makes perfect sense. My aim with the independent lines was to have the raise inside the service function so that the backtrace does not go outside the service. However, if we make the helper func take a Response as @elliotgunton suggested, then we get the best of both worlds:
- isolated func + code
- users going through a backtrace still get to only focus on the service function because that's the spot that raises